test/e2e/migrationhierarchy_test.go

```diff
@@ -0,0 +1,341 @@
+		FieldShouldContain("namespace", "", nsG, ".metadata.labels", danav1.Parent+":"+nsF)
+
+		FieldShouldContain("subnamespace", nsG, nsH, ".metadata.namespace", nsG)
+		FieldShouldContain("namespace", "", nsH, ".metadata.labels", danav1.Parent+":"+nsG)
+
+		// add test label to the migrated ns
+		LabelTestingNs(nsF)
+		LabelTestingNs(nsG)
+		LabelTestingNs(nsH)
+	})
+
+	It("should not migrate an empty resourcepool to be under subnamespace, should get error phase", func() {
+		nsA := GenerateE2EName("a")
+		nsB := GenerateE2EName("b")
+		nsC := GenerateE2EName("c")
+		nsD := GenerateE2EName("d")
+		nsE := GenerateE2EName("e")
+		nsF := GenerateE2EName("f")
+		nsG := GenerateE2EName("g")
+
+		// create hierarchy
+		CreateSubnamespace(nsA, nsRoot, false, storage, "50Gi", cpu, "50", memory, "50Gi", pods, "50", gpu, "50")
+		CreateSubnamespace(nsB, nsA, false, storage, "25Gi", cpu, "25", memory, "25Gi", pods, "25", gpu, "25")
+		CreateSubnamespace(nsC, nsB, false, storage, "10Gi", cpu, "10", memory, "10Gi", pods, "10", gpu, "10")
+		CreateSubnamespace(nsD, nsC, false, storage, "5Gi", cpu, "5", memory, "5Gi", pods, "5", gpu, "5")
+		CreateSubnamespace(nsE, nsC, false, storage, "5Gi", cpu, "5", memory, "5Gi", pods, "5", gpu, "5")
+		CreateSubnamespace(nsF, nsD, true, storage, "2Gi", cpu, "2", memory, "2Gi", pods, "2", gpu, "2")
+		CreateSubnamespace(nsG, nsF, true)
+
+		// create the migration hierarchy and return name to mhname
+		mhname := CreateMigrationHierarchy(nsG, nsE)
+
+		// make sure the migration hierarchy was created in error phase and did not migrate the subnamespace
+		FieldShouldContain("subnamespace", nsF, nsG, ".metadata.namespace", nsF)
+		FieldShouldContain("namespace", "", nsG, ".metadata.labels", danav1.Parent+":"+nsF)
+		FieldShouldContain("migrationhierarchy", "", mhname, ".status.phase", "Error")
+	})
+
+	It("should not migrate subnamespaces between secondary root", func() {
+		nsA := GenerateE2EName("a")
+		nsB := GenerateE2EName("b")
+		nsC := GenerateE2EName("c")
+		nsD := GenerateE2EName("d")
+		nsE := GenerateE2EName("e")
+
+		// create hierarchy
+		CreateSubnamespace(nsA, nsRoot, false, storage, "50Gi", cpu, "50", memory, "50Gi", pods, "50", gpu, "50")
+		CreateSubnamespace(nsB, nsRoot, false, storage, "50Gi", cpu, "50", memory, "50Gi", pods, "50", gpu, "50")
+		CreateSubnamespace(nsC, nsA, false, storage, "25Gi", cpu, "25", memory, "25Gi", pods, "25", gpu, "25")
+		CreateSubnamespace(nsD, nsC, false, storage, "10Gi", cpu, "10", memory, "10Gi", pods, "10", gpu, "10")
+		CreateSubnamespace(nsE, nsD, false, storage, "5Gi", cpu, "5", memory, "5Gi", pods, "5", gpu, "5")
+
+		// make sure the subnamespace was not migrated and the parent has not been updated
+		ShouldNotCreateMigrationHierarchy(nsE, nsB)
+		FieldShouldContain("subnamespace", nsD, nsE, ".metadata.namespace", nsD)
+		FieldShouldContain("namespace", "", nsE, ".metadata.labels", danav1.Parent+":"+nsD)
+	})
```

Comment:
Where is the secondary root check? [See here](https://github.com/dana-team/hns/blob/main/test/e2e/updatequota_test.go#L189-L190)

Reply:
Is it later on the file. this one is old, I removed it.